### PR TITLE
Fix PHP 8.2 deprecation warnings

### PR DESCRIPTION
**Pull request by TomA-R**
tl;dr: On PHP 8.2, this package produces some warnings. This PR fixes that, and makes sure unit tests can be run on modern PHP versions too.

### Address PHP 8.2 deprecation warnings
Fix deprecation warnings like:
```
PHP Deprecated:  Creation of dynamic property mersenne_twister\twister::$bits32 is deprecated in php-mersenne-twister/src/mersenne_twister.php on line 59
PHP Deprecated:  Creation of dynamic property mersenne_twister\twister::$mt is deprecated in php-mersenne-twister/src/mersenne_twister.php on line 69
PHP Deprecated:  Creation of dynamic property mersenne_twister\twister::$mti is deprecated in php-mersenne-twister/src/mersenne_twister.php on line 70
```

These show up on PHP 8.2 - see [this](https://php.watch/versions/8.2/dynamic-properties-deprecated).

### Set minimum PHP version to 7.3 and upgrade PHPUnit and related dependencies

That's because the current versions of dependencies require old versions of PHP.

```
$ composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - doctrine/instantiator is locked to version 1.3.0 and an update of this package was not requested.
    - doctrine/instantiator 1.3.0 requires php ^7.1 -> your php version (8.2.1) does not satisfy that requirement.
  Problem 2
    - myclabs/deep-copy is locked to version 1.9.5 and an update of this package was not requested.
    - myclabs/deep-copy 1.9.5 requires php ^7.1 -> your php version (8.2.1) does not satisfy that requirement.
```

### Fix unit test
`mersenne_twister\TwisterTest::testFailureOnNonExistentInitFile` was failing because modern versions of PHP don't accept negative values to be passed to file_get_contents() as "length":

```
1) mersenne_twister\TwisterTest::testFailureOnNonExistentInitFile
ValueError: file_get_contents(): Argument #5 ($length) must be greater than or equal to 0

php-mersenne-twister/src/mersenne_twister.php:146
php-mersenne-twister/src/mersenne_twister.php:163
php-mersenne-twister/tests/phpunit/tests/TwisterTest.php:50
```

## Testing
Tests still pass across PHP versions
```
$ ./vendor/bin/phpunit -c phpunit.xml
PHPUnit 9.6.5 by Sebastian Bergmann and contributors.

......                                                              6 / 6 (100%)

Time: 00:00.008, Memory: 6.00 MB

OK (6 tests, 6 assertions)
```

ping @ruafozy

### CODE REVIEW

**Comment by ruafozy** [comment]
Thanks. I should have time to look at this at the weekend.
